src/main/java/com/kryptnostic/datastore/services/UserDirectoryService.java

```diff
@@ -1,64 +1,57 @@
 package com.kryptnostic.datastore.services;
 
+import com.dataloom.client.RetrofitFactory;
 import com.dataloom.directory.pojo.Auth0UserBasic;
 import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
+import com.kryptnostic.datastore.exceptions.BadRequestException;
 import org.json.simple.JSONObject;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.http.MediaType;
-import retrofit.RequestInterceptor;
-import retrofit.RestAdapter;
+import retrofit2.Retrofit;
 
 import java.util.List;
 import java.util.Map;
 
 public class UserDirectoryService {
 
-    private static final Logger logger = LoggerFactory.getLogger( UserDirectoryService.class );
-
-    private RestAdapter        adapter;
+    private Retrofit           retrofit;
     private Auth0ManagementApi auth0ManagementApi;
 
     public UserDirectoryService( String token ) {
-        adapter = new RestAdapter.Builder()
-                .setEndpoint( "https://loom.auth0.com/api/v2" )
-                .setRequestInterceptor(
-                        (RequestInterceptor) facade -> {
-                            facade.addHeader( "Authorization", "Bearer " + token );
-                            facade.addHeader( "Content-Type", MediaType.APPLICATION_JSON_VALUE );
-                        } )
-                .setLogLevel( RestAdapter.LogLevel.FULL )
-                .setLog( msg -> logger.debug( msg.replaceAll( "%", "[percent]" ) ) )
-                .build();
-        auth0ManagementApi = adapter.create( Auth0ManagementApi.class );
+        retrofit = RetrofitFactory.newClient( "https://loom.auth0.com/api/v2/", () -> token );
+        auth0ManagementApi = retrofit.create( Auth0ManagementApi.class );
     }
 
     public Map<String, Auth0UserBasic> getAllUsers() {
         Map<String, Auth0UserBasic> res = Maps.newHashMap();
         JSONObject[] jsonObjects = auth0ManagementApi.getAllUsers();
 
         for ( int i = 0; i < jsonObjects.length; i++ ) {
-            Auth0UserBasic auth0UserBasic = parsingToPOJO( jsonObjects[i] );
+            Auth0UserBasic auth0UserBasic = parsingToPOJO( jsonObjects[ i ] );
             res.put( auth0UserBasic.getUserId(), auth0UserBasic );
         }
         return res;
     }
 
     public Auth0UserBasic getUser( String userId ) {
         JSONObject jsonObject = auth0ManagementApi.getUser( userId );
+        if ( jsonObject == null ) {
+            throw new BadRequestException( "User does not exist!" );
+        }
         return parsingToPOJO( jsonObject );
     }
 
-    private Auth0UserBasic parsingToPOJO( JSONObject jsonObject ){
+    private Auth0UserBasic parsingToPOJO( JSONObject jsonObject ) {
```

Comment:
You don't need this anymore just return an Auth0UserBasic in your retrofit interface.